### ISSUE TITLE
How can I get the entropy of the output of the segmentation?

### ISSUE THREAD

**Issue by zh994** [Chen-Ziyang/VPTTA#5]
Thanks for your great work. I can get results similar to those recorded in the paper using your method, but I get very poor results when using the tent . Could you provide some details on using the tent , such as the learning rate, the calculation of entropy, etc.?

**Comment by zh994** [comment]
I used the following code to calculate entropy and set the learning rate to 1e-4 to achieve results similar to those in the paper。
`def forward_and_adapt(x, model, optimizer):
    """Forward and adapt model on batch of data.

    Measure entropy of the model prediction, take gradients, and update params.
    """
    # forward
    outputs, fea = model(x)
    # adapt
    logits = torch.sigmoid(outputs)
    loss = (-(torch.log(logits) * logits).sum(1)).mean()
    loss.backward()
    optimizer.step()
    optimizer.zero_grad()
    return outputs
`

**Comment by Chen-Ziyang** [comment]
In our implementation of TENT, we used codes similar to your provided ones and employed the Adam optimizer with a learning rate of 1e-4. 
Besides the entropy calculation, there are two key implementation details in TENT.
1) Only calculate and backpropagate the gradients of the weight and bias within each BN layer.
2) Set both the running mean and running variance to None and perform inference on the test data using the "train" mode.

Please note that our reproduction slightly differs from the original TENT. After training the model with test data A, we use the fine-tuned model to perform inference on A. This implies that each test data undergoes two forward calculations. Since our VPTTA also requires two forward calculations, we implemented this to ensure fairness.

**Comment by zh994** [comment]
Thank you for your reply.
